pkg/state/state_sriov_dp_test.go

```diff
@@ -14,130 +14,280 @@ See the License for the specific language governing permissions and
-	image := imageSpec.Repository + "/" + imageSpec.Image + ":" + imageSpec.Version
-	template := obj.Object["spec"].(map[string]interface{})["template"].(map[string]interface{})
-	jsonSpec, _ := obj.MarshalJSON()
-	spec := fmt.Sprintf("%v", template)
-
-	Expect(obj.GetKind()).To(Equal("DaemonSet"))
-	Expect(obj.Object["metadata"].(map[string]interface{})["namespace"].(string)).To(Equal(namespace))
-	Expect(spec).To(ContainSubstring(image))
-	Expect(string(jsonSpec)).To(ContainSubstring(nodeAffinity))
-}
-
 var _ = Describe("SR-IOV Device Plugin State tests", func() {
+	var (
+		sriovDpState state.State
+		catalog      state.InfoCatalog
+		client       client.Client
+		namespace    string
+		renderer     state.ManifestRenderer
+	)
 
-	Context("When creating NCP with SRIOV-device-plugin", func() {
-		It("Should Apply", func() {
-			client := mocks.ControllerRuntimeClient{}
-			manifestBaseDir := "../../manifests/state-sriov-device-plugin"
+	BeforeEach(func() {
+		scheme := runtime.NewScheme()
+		Expect(mellanoxv1alpha1.AddToScheme(scheme)).NotTo(HaveOccurred())
+		Expect(v1.AddToScheme(scheme)).NotTo(HaveOccurred())
+		Expect(appsv1.AddToScheme(scheme)).NotTo(HaveOccurred())
+		client = fake.NewClientBuilder().WithScheme(scheme).Build()
+		manifestDir := "../../manifests/state-sriov-device-plugin"
+		s, r, err := state.NewStateSriovDp(client, manifestDir)
+		Expect(err).NotTo(HaveOccurred())
+		sriovDpState = s
+		renderer = r
+		catalog = getTestCatalog()
+		namespace = config.FromEnv().State.NetworkOperatorResourceNamespace
+	})
 
-			files, err := utils.GetFilesWithSuffix(manifestBaseDir, render.ManifestFileSuffix...)
+	Context("When creating NCP with SRIOV-device-plugin", func() {
+		It("should create Daemonset - minimal spec", func() {
+			By("Sync")
+			cr := getMinimalNicClusterPolicyWithSriovDp(false)
+			status, err := sriovDpState.Sync(context.Background(), cr, catalog)
 			Expect(err).NotTo(HaveOccurred())
-			renderer := render.NewRenderer(files)
-
-			stateName := "state-SRIOV-device-plugin"
-			sriovDpState := stateSriovDp{
-				stateSkel: stateSkel{
-					name:        stateName,
-					description: "SR-IOV device plugin deployed in the cluster",
-					client:      &client,
-					renderer:    renderer,
-				},
+			Expect(status).To(BeEquivalentTo(state.SyncStateNotReady))
+			By("Verify DaemonSet")
+			ds := &appsv1.DaemonSet{}
+			err = client.Get(context.Background(), types.NamespacedName{Namespace: namespace,
+				Name: "sriov-device-plugin"}, ds)
+			Expect(err).NotTo(HaveOccurred())
+			assertCommonDaemonSetFields(ds, &cr.Spec.SriovDevicePlugin.ImageSpec, cr)
+			// expect privileged mode
+			Expect(*ds.Spec.Template.Spec.Containers[0].SecurityContext.Privileged).To(BeTrue())
+			assertSriovDpPodTemplatesVolumeFields(&ds.Spec.Template, false)
+			assertSriovDpPodTemplatesVolumeMountFields(&ds.Spec.Template, false)
+		})
+		It("should create Daemonset with CDI support when specified in CR", func() {
```

Comment:
can you add a check that device plugin is started with "--use-cdi" flag ? unless i missed it in the test :)

Reply:
That is what this test is about. `cr := getMinimalNicClusterPolicyWithSriovDp(true)` sets `useCDI = true` in the devicePlugin spec.

Reply:
please show me the line where the test check that "--use-cdi" flag is passed to device plugin command in the daemonset.

Reply:
Ok, I just added it.